Assets/Mirror/Core/NetworkBehaviourHybrid.cs

```diff
@@ -0,0 +1,479 @@
+        public bool baselineIsDelta = true;
+
+        // change detection: we need to do this carefully in order to get it right.
+        //
+        // DONT just check changes in UpdateBaseline(). this would introduce MrG's grid issue:
+        //   server start in A1, reliable baseline sent to client
+        //   server moves to A2, unreliabe delta sent to client
+        //   server moves to A1, nothing is sent to client becuase last baseline position == position
+        //   => client wouldn't know we moved back to A1
+        //
+        // INSTEAD: every update() check for changes since baseline:
+        //   UpdateDelta() keeps sending only if changed since _baseline_
+        //   UpdateBaseline() resends if there was any change in the period since last baseline.
+        //   => this avoids the A1->A2->A1 grid issue above
+        bool changedSinceBaseline = false;
+
+        [Header("Debug")]
+        public bool debugLog = false;
+
+        public virtual void Reset()
+        {
+            lastSerializedBaselineTick = 0;
+            lastDeserializedBaselineTick = 0;
+            changedSinceBaseline = false;
+        }
+
+        // user callbacks //////////////////////////////////////////////////////
+        protected abstract void OnSerializeBaseline(NetworkWriter writer);
+        protected abstract void OnDeserializeBaseline(NetworkReader reader, byte baselineTick);
+
+        protected abstract void OnSerializeDelta(NetworkWriter writer);
+        protected abstract void OnDeserializeDelta(NetworkReader reader, byte baselineTick);
+
+        // implementations must store the current baseline state when requested:
+        // - implementations can use this to compress deltas against
+        // - implementations can use this to detect changes since baseline
+        // this is called whenever a baseline was sent.
+        protected abstract void StoreState();
+
+        // implementations may compare current state to the last stored state.
+        // this way we only need to send another reliable baseline if changed since last.
+        // this is called every syncInterval, not every baseline sync interval.
+        // (see comments where this is called).
+        protected abstract bool StateChanged();
+
+        // user callback in case drops due to baseline mismatch need to be logged/visualized/debugged.
+        protected virtual void OnDrop(byte lastBaselineTick, byte baselineTick, NetworkReader reader) {}
+
+        // rpcs / cmds /////////////////////////////////////////////////////////
+        [ClientRpc(channel = Channels.Reliable)] // reliable baseline
+        void RpcServerToClientBaseline(ArraySegment<byte> data)
+        {
+            // baseline is broadcast to all clients.
+            // ignore if this object is owned by this client.
+            if (IsClientWithAuthority) return;
```

Comment:
Use `includeOwner = false` or add comments why not.

Reply:
done